### ISSUE TITLE
Implement the functionality of legacy CC Search.

### ISSUE THREAD

**Issue by kgodey** [creativecommons/cccatalog-frontend#315]
**Is your feature request related to a problem? Please describe.**
We would like to decommission legacy CC Search (https://oldsearch.creativecommons.org/) as soon as possible but it does offer some functionality that the new CC Search does not – the ability to search CC licensed works that are not images. We would like to integrate that functionality into the current site.

**Describe the solution you'd like**
- A search bar that takes a search query and provides the user with a list of providers to select from. When the user searches, the site should redirect the user to the search results page of that provider with the query already filled in.
- We should only show providers that provide content types that are not images. For images, we want them to use the primary CC Search experience.
- It should be **really clear** that this is not the primary CC Search experience. One way to do this is to put this functionality in a separate page linked to from the top nav. Feel free to propose another idea.

**Additional context**
Since this is a major addition, the implementation needs product input from CC staff. Please post your design here before implementing the full functionality so that we can give you feedback.

**Comment by annatuma** [comment]
@kgodey I'm moving this to the Parking Lot, since we may simply prioritize integrations with those collections directly through (new) CC Search instead. I'll add those to Catalog Pipeline for further research. Definitely move back if you have a different view on this.

**Comment by kgodey** [comment]
@annatuma I don't think we can integrate stuff like Google Images into CC Search. I think we need some sort of plan to shut down old CC Search; we shouldn't keep maintaining it forever. I'm worried moving it into the Parking Lot will mean we forget about it.

**Comment by Dhruvi16** [comment]
I would like to take it up. 
Thank you.

**Comment by genedy377** [comment]
hey @kgodey I think that i have skills that  can help with this issue can i work on it ?

**Comment by akmadian** [comment]
@genedy377 go ahead! Please remember to comment on this issue with your plans for adding the functionality so CC staff can give feedback

**Comment by abhisheknaiidu** [comment]
Hi! @akmadian @kgodey Can i work on this issue, if no one is working on this !

**Comment by kgodey** [comment]
I think we should run this by @annatuma, this would be a significant UI and functionality change.

**Comment by annatuma** [comment]
Hey folks, apologies for the delayed response here. I've gone ahead and blocked this one. This will require thoughtful internal discussion, and hasn't been prioritized to date because of other work on (new) CC Search.

@genedy377 @abhisheknaiidu please look for other "help wanted" issues to contribute to.

**Comment by annatuma** [comment]
We're currently mocking up what this functionality could look like on new search. Under consideration is [content type below search bar](https://app.moqups.com/OD23yiSWK9/view/page/a854244a3), [content type in navbar](https://app.moqups.com/OD23yiSWK9/view/page/ad1710be3), further options to be discussed and considered.

Working notes, regarding Sources:

**Europeana** image integration is forthcoming. We can include them on audio & video pages on CC Search with filters in place for "TYPE"

Sample query from Old Search:
https://www.europeana.eu/en/search?query=dogs%20AND%20RIGHTS%3A%2acreative%2a%20AND%20NOT%20RIGHTS%3A%2anc%2a%20AND%20NOT%20RIGHTS%3A%2and%2a

Sample query with video filter:
https://www.europeana.eu/en/search?page=1&qf=TYPE%3A%22VIDEO%22&query=dogs%20AND%20RIGHTS%3A%2acreative%2a%20AND%20NOT%20RIGHTS%3A%2anc%2a%20AND%20NOT%20RIGHTS%3A%2and%2a&view=undefined

Sample query with audio filter:
https://www.europeana.eu/en/search?page=1&qf=TYPE%3A%22SOUND%22&query=dogs%20AND%20RIGHTS%3A%2acreative%2a%20AND%20NOT%20RIGHTS%3A%2anc%2a%20AND%20NOT%20RIGHTS%3A%2and%2a&view=undefined

**Flickr** is indexed by CC Search

**Google (web)** currently filters by "as_rights"

Sample search query:
https://www.google.com/search?as_rights=(cc_publicdomain|cc_attribute|cc_sharealike).-(cc_noncommercial|cc_nonderived)&q=dogs&hl=

We may include information on how to filter on Google by resource rights on a help page on CC Search, but this likely won't be built into the CC Search UI prominently.

**Google Images** will not be indexed by CC Search. At present, similar to Google (web), we may reference how to search Google Images by open content, but won't support it in the UI.

**Jamendo** have changed their search functionality, so the current experience linked from Old Search doesn't work. Assuming we can figure out how to filter by CC-licensed music again, we'd include this on an Audio page.

**Open Clip Art Library** will hopefully be indexed on CC Search as an image source (https://github.com/creativecommons/cccatalog/issues/169). Until then, we may include them in a resource list of non-indexed image providers, but this would be secondary to UI support for video/audio from old Search.

**SpinXPress** is offline.

**Wikimedia Commons** as a source has both audio and video available that we'd like to allow users to search, respectively. We'd need to add the parameter "filetype" for each.

Sample audio search:
https://commons.wikimedia.org/w/index.php?sort=relevance&search=dogs+filetype%3Aaudio&title=Special%3ASearch&profile=advanced&fulltext=1&advancedSearch-current=%7B%22fields%22%3A%7B%22filetype%22%3A%22audio%22%7D%7D&ns0=1&ns6=1&ns14=1

Sample video search:
https://commons.wikimedia.org/w/index.php?sort=relevance&search=dogs+filetype%3Avideo&title=Special%3ASearch&profile=advanced&fulltext=1&advancedSearch-current=%7B%22fields%22%3A%7B%22filetype%22%3A%22video%22%7D%7D&ns0=1&ns6=1&ns14=1

**YouTube** won't be indexed in the long run, most likely, similar to Google Images. In the short term we can support this source in a "non-indexed" UI for video. If/when we do index video sources, we'll need to transfer this to either a resource page (like Google Images) or find another way to support pre-filtered search for YouTube.

Sample query:
https://www.youtube.com/results?search_query=dogs%2Ccreativecommons

**Pixabay** no longer supports CC licenses for content, and won't be indexed on CC Search for images.

**ccMixter** will be added to the queue for audio indexing on CC Search. In the interim, they should be supported on the audio meta search UI.

Sample search query:
http://ccmixter.org/api/query?datasource=uploads&search_type=all&sort=rank&search=dogs&lic=by,sa,s,splus,pd,zero

**SoundCloud** is in the queue for audio indexing on CC Search. In the interim, they should be supported on the audio meta search UI.

Sample query:
https://soundcloud.com/search/sounds?q=dogs&filter.license=to_modify_commercially

Filters include "to modify commercially" and "to share commercially" so we need to ensure we map to those filters depending on the filters set in our own UI.

In summary, we plan to support the following sources for meta search of audio and video, until we're able to index those content types:

**Audio meta search**
Europeana
Wikimedia Commons
Jamendo
ccMixter
SoundCloud

**Video meta search**
Europeana
Wikimedia Commons
YouTube

**Comment by annatuma** [comment]
Start with a fresh meta ticket for this: https://github.com/creativecommons/cccatalog-frontend/issues/921 and two sub tickets to tackle each addition.

Closing this one out.
